File: cpp/Examples/bubblesort.cpp
#include <stdlib.h>  
#include <time.h>   
#include "visualarray.h"

void bubbleSort(int* array, int n) {
	for (int i=n-1; i>=0; i--) {
		for (int j=0; j<=n; j++) {
			if (array[j] > array[j+1]) {
				int temp = array[j];
				array[j] = array[j+1];
				array[j+1] = temp;
			}
		}
	}
}

int main() {
	const int SIZE = 10;
	VisualArray<int> varray = VisualArray(SIZE);
	srand(time(NULL));
	for (int i=0; i<SIZE; i++) {
		varray[i] = (rand() % 40) + 10;
	}
	bubbleSort(varray, SIZE);
	varray.render();

	return 0;
}

File: python2/Examples/bubblesort.py
import sys
sys.path.insert(0,'..')
from visualarray import VisualArray
import random

def bubblesort(alist):
    for num in range(len(alist)-1,0,-1):
        for i in range(num):
            if alist[i]>alist[i+1]:
            	alist[i], alist[i+1] = alist[i+1], alist[i]

alist = random.sample(xrange(10,50), 10)
alist = VisualArray(alist)
bubblesort(alist)
alist.render()
